library/src/amd_detail/rocblaslt/src/include/rocblaslt_mat_utils.hpp

```diff
@@ -382,6 +382,11 @@ inline rocblaslt_status rocblaslt_matmul_valid_args(const rocblaslt_matmul_desc
     n = num_cols_d;
     k = (opA == HIPBLAS_OP_N) ? num_cols_a : num_rows_a;
 
+    if(compute_type == rocblaslt_compute_f16) {
```

Comment:
have you checked the output of HIPBLASLT_LOG_MASK=32 for compute_f16 case?
The output should be identical with what user set.